src/stores/main/utils.ts

```diff
@@ -18,3 +19,250 @@ export function mapOAuth2UserToUser(user: IOAuth2User): ILoggedInUser {
+      };
+    }
+
+    const wildcardPermissionsKeys: string[] = [];
+
+    if (group.length > 0) {
+      wildcardPermissionsKeys.push(hashPermissionKey('*', '*', '*'));
+    }
+
+    if (resource.length > 0) {
+      wildcardPermissionsKeys.push(hashPermissionKey(group, '*', '*'));
+    }
+
+    if (resourceName.length > 0) {
+      wildcardPermissionsKeys.push(hashPermissionKey(group, resource, '*'));
+    }
+
+    /**
+     * Merge permissions from  wildcard rules that include all (`*`)
+     * groups, resources and resource names with the ones
+     * that include specific groups, resources and resource names.
+     *
+     * For example: add permissions from `:apps:*` to `:apps:some-app`.
+     * @example `*:*:*` and `apps.gs.io:apps:*`
+     * @example `apps.gs.io:*:*` and `apps.gs.io:apps:*`
+     * @example `:apps:*` and `:apps:some-app`
+     */
+    for (const wildcardKey of wildcardPermissionsKeys) {
+      if (newPermissions.hasOwnProperty(wildcardKey)) {
+        appendKeyToNamespacePermissions(
+          key,
+          newPermissions[wildcardKey],
+          newPermissions
+        );
+      }
+    }
+  }
+
+  return newPermissions;
+}
+
+/**
+ * Check if the user has the permission to use
+ * the given verb for the given resource configuration.
+ * @param permissions
+ * @param verb
+ * @param group
+ * @param resource
+ * @param resourceName
+ */
+export function hasNamespacePermission(
+  permissions: INamespacePermissions,
+  verb: PermissionVerb,
+  group: string,
+  resource: string,
+  resourceName: string = '*'
+): boolean {
+  let verbs: string[] = [];
+
+  const key = hashPermissionKey(group, resource, resourceName);
+  if (permissions.hasOwnProperty(key)) {
+    verbs = permissions[key];
+  } else {
+    const wildcardKey = hashPermissionKey('*', '*', '*');
+    if (!permissions.hasOwnProperty(wildcardKey)) return false;
+
+    verbs = permissions[wildcardKey];
+  }
+
+  if (verbs.length === 1 && verbs[0] === '*') return true;
+
+  return verbs.includes(verb);
+}
+
+/**
+ * Check if the user has the permission to use
+ * the given verb for the given resource configuration
+ * in the given namespace.
+ * @param permissions
+ * @param namespace
+ * @param verb
+ * @param group
+ * @param resource
+ * @param resourceName
+ */
+export function hasPermission(
+  permissions: IPermissionMap,
+  namespace: string,
+  verb: string,
+  group: string,
+  resource: string,
+  resourceName?: string
+): boolean {
+  if (!permissions.hasOwnProperty(namespace)) return false;
+
+  return hasNamespacePermission(
+    permissions[namespace],
+    verb,
+    group,
+    resource,
+    resourceName
+  );
+}
+
+/**
+ * Compute an organization namespace from the given organization name.
+ * This also makes the org name follow the [DNS label standard](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names).
+ * @param name
+ */
+export function getNamespaceFromOrgName(name: string): string {
```

Comment:
This is smarter now, by transforming the organization name into a DNS friendly format